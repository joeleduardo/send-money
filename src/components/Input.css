input {
  margin-top: .5rem;
  padding: .5rem 0;
  border: none;
  border-bottom: 1px solid #cccccc;
  font-size: 1rem;
  width: 100%;
  transition: border-color ease-in-out .15s;
  box-sizing: border-box;
}

input:active,
input:focus {
   outline: transparent;
   border-color: #4b3cfa;
}

label {
  padding: .5rem;
  font-size: .9rem;
  color: #999999;
}

.currency {
  position: absolute;
  top: 2.5rem;
  left: .5rem;
}

.input {
  position: relative;
  display: flex;
  flex-direction: column;
  margin-bottom: 1rem;
}

.error-msg {
  color: #F1586B;
  padding: .5rem;
  font-size: .8rem;
}