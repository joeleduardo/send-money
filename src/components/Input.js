import React from 'react';
import PropTypes from 'prop-types';
import './Input.css';

const Input = (props) => {
  const {
    label,
    name,
    value,
    prompt,
    validation,
    currency,
    onChange
  } = props;

  const currencyStyle = currency ? { paddingLeft: '1rem' } : {};
  const validationStyle = validation ? { borderColor: '#F1586B' } : {};

  return (
    <div className='input'>
      {currency && <div className='currency'>{currency}</div>}
      <label htmlFor={name}>
        {label}
        <input style={{...currencyStyle, ...validationStyle}} id={name} name={name} type='text' value={value || ''} placeholder={prompt} onChange={e => onChange(e)} autoComplete='off' />
      </label>
      {validation && <div className='error-msg'>{validation}</div>}
    </div>
  );
};

Input.propTypes = {
  name: PropTypes.string.isRequired,
  label: PropTypes.string.isRequired,
  prompt: PropTypes.string,
  value: PropTypes.string,
  validation: PropTypes.string,
  currency: PropTypes.string,
  onChange: PropTypes.func
};

Input.defaultProps = {
  name: 'inputField',
  label: 'My field',
  prompt: '',
  value: '',
  validation: '',
  currency: ''
};

export default Input;
