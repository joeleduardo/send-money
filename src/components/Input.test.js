import React from 'react';
import { shallow } from 'enzyme';
import Input from './Input';

const event = {
  target: { name: 'name', value: 'Joel Ortiz' }
};

test('Input component', () => {
  const mockChange = jest.fn();
  const cInput = shallow(<Input name='name' onChange={(e) => mockChange(e)} label='Name' prompt='Friend name' validation='' currency='' value='' />);
  cInput.find('input').simulate('change', event);
  expect(mockChange).toBeCalledWith(event);
  expect(cInput).toMatchSnapshot();
});

test('Input component currency, validation', () => {
  const cInput = shallow(<Input name='name' label='Name' prompt='Friend name' validation='Please enter a valid name' currency='â‚¬' />);
  expect(cInput).toMatchSnapshot();
});