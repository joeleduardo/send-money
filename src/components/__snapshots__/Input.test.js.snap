// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Input component 1`] = `
<div
  className="input"
>
  <label
    htmlFor="name"
  >
    Name
    <input
      autoComplete="off"
      id="name"
      name="name"
      onChange={[Function]}
      placeholder="Friend name"
      style={Object {}}
      type="text"
      value=""
    />
  </label>
</div>
`;

exports[`Input component currency, validation 1`] = `
<div
  className="input"
>
  <div
    className="currency"
  >
    â‚¬
  </div>
  <label
    htmlFor="name"
  >
    Name
    <input
      autoComplete="off"
      id="name"
      name="name"
      onChange={[Function]}
      placeholder="Friend name"
      style={
        Object {
          "borderColor": "#F1586B",
          "paddingLeft": "1rem",
        }
      }
      type="text"
      value=""
    />
  </label>
  <div
    className="error-msg"
  >
    Please enter a valid name
  </div>
</div>
`;
